File: product-processor-service/src/main/java/com/example/product_processor_service/service/product/grpc/client/ProductGrpcClientService.java
package com.example.product_processor_service.service.product.grpc.client;

import com.example.grpc.user.ProductProtoServiceGrpc;
import com.example.grpc.user.UserProtoConfiguration;
import com.example.product_processor_service.service.common.grpc.client.GrpcClientService;
import com.example.product_processor_service.util.mapper.UserGrpcMapper;

public abstract class ProductGrpcClientService extends GrpcClientService<ProductProtoServiceGrpc.ProductProtoServiceBlockingStub> {
    public ProductGrpcClientService(ProductProtoServiceGrpc.ProductProtoServiceBlockingStub stub) {
        super(stub);
    }

    public void save(UserProtoConfiguration.ProductRegistrationMessage productMessage) {
        stub.save(productMessage);
    }

    public void deleteAccountProductByAccountUuidAndProductUri(UserProtoConfiguration.AccountUuidProductUriMessage accountUuidProductUriMessage) {
        stub.deleteAccountProductByAccountUuidAndProductUri(accountUuidProductUriMessage);
    }

    public void update(UserProtoConfiguration.ProductMessage productMessage) {
        stub.update(productMessage);
    }

    public UserProtoConfiguration.ProductMessage getByUrl(UserProtoConfiguration.StringMessage stringMessage) {
        return stub.getByUrl(stringMessage);
    }

    public UserProtoConfiguration.ProductListMessage getAllExpired() {
        return stub.getAllExpired(UserGrpcMapper.mapTo());
    }

    public UserProtoConfiguration.ProductListMessage getAllByAccountUuidAndPage(UserProtoConfiguration.StringIntMessage stringIntMessage) {
        return stub.getAllByAccountUuidAndPage(stringIntMessage);
    }

    public UserProtoConfiguration.BooleanMessage existsByUrl(UserProtoConfiguration.StringMessage stringMessage) {
        return stub.existsByUrl(stringMessage);
    }
}

File: product-manager-service/src/main/kotlin/com/example/product_manager_service/service/product/grpc/ProductGrpcClientService.kt
package com.example.product_manager_service.service.product.grpc

import com.example.grpc.user.ProductProtoServiceGrpc
import com.example.grpc.user.UserProtoConfiguration
import com.example.product_manager_service.service.common.grpc.GrpcClientService

abstract class ProductGrpcClientService(stub: ProductProtoServiceGrpc.ProductProtoServiceBlockingStub) :
    GrpcClientService<ProductProtoServiceGrpc.ProductProtoServiceBlockingStub>(stub) {

    fun getAllByAccountUuid(message: UserProtoConfiguration.StringIntMessage): UserProtoConfiguration.ProductListMessage =
        stub.getAllByAccountUuidAndPage(message)

}